includes/wikia/GlobalTitle.php

```diff
@@ -70,6 +70,12 @@ public static function newFromText( $text, $namespace = NS_MAIN, $city_id = null
 		}
 
 		$filteredText = Sanitizer::decodeCharReferences( $text );
+
+		if ( $namespace === NS_SPECIAL ) {
+			$localName = SpecialPageFactory::getLocalNameFor( $text );
```

Comment:
This is called before the content language of the target wiki is set in `loadAll`, so will be in the language of the wiki this is requested on rather than the target wiki, i.e. (English Wookieepedia is 147, and Japanese Wookieepedia is 5931):
```
$ SERVER_ID=147 php eval.php 
> $gt = GlobalTitle::newFromText( 'Forum', NS_SPECIAL, 5931 );

> echo $gt->getFullURL();
http://ja.starwars.grunny.wikia-dev.com/wiki/%E7%89%B9%E5%88%A5:Forum

$ SERVER_ID=5931 php eval.php 
> $gt = GlobalTitle::newFromText( 'Forum', NS_SPECIAL, 147 );

> echo $gt->getFullURL();
http://starwars.grunny.wikia-dev.com/wiki/Special:%E3%83%95%E3%82%A9%E3%83%BC%E3%83%A9%E3%83%A0
```